uart.py

```diff
@@ -4,65 +4,162 @@
 import time
 import threading
 
-import perspective
+from .perspective import PerspectiveTransform
 
 import cv2
 
-#sys.path.append(os.path.join(os.path.dirname(_file_), '../../..'))
+from PIL import Image
+
+# sys.path.append(os.path.join(os.path.dirname(_file_), '../../..'))
 from uarm.wrapper import SwiftAPI
 from uarm.tools.list_ports import get_ports
 
-class uart_api:
-    available_pixel = {} #rgb values of all the paints
-    swift = None #robot arm object
-    device_info = None 
+
+class uart:
+
+    # rgb values of all the paints
+    available_pixel = {}
+
+    # robot arm object
+    swift = None
+    device_info = None
     firmware_version = None
-    image = None #image you're trying to paint
-    canvas = None #image of the canvas as you're working on it
-    canvas_corners = None #points of the four corners of the canvas (in robot arm coords)
-    ptransform = None #contains the warped image of
-    M = None #transformation matrix
-    def __init__(self, im):
-        self.available_pixel = {'red':[255,0,0], 'green':[0,255,0], 'blue':[0,0,255],'magenta':[255,0,255], 'tomato':[255,99,71], 'lawn green':[124,252,0]}
-        self.swift = SwiftAPI(filters={'hwid': 'USB VID:PID=2341:0042'})
-        self.device_info = self.swift.get_device_info()
-        self.firmware_version = self.device_info['firmware_version']
-        self.swift.set_mode(0)
-        self.image = im
-        print("Setting four corners; input tl, tr, bl or br")
-        self.canvas_corners = self.setFourCorners()
-        _, cap = cv2.VideoCapture(0).read()
-        self.ptransform = perspective.PerspectiveTransform(self.image)        
-        self.M = self.get_m()
+
+    # image you're trying to paint
+    image = None
+
+    # image of the canvas as you're working on it
+    canvas = None
+
+    # points of the four corners of the canvas (in robot arm coords)
+    canvas_corners = None
+
+    # contains the warped image of
+    ptransform = None
+
+    # transformation matrix
+    M = None
```

Comment:
could probably scrap this alongside the methods that use it.

Reply:
The M function?